### ISSUE TITLE
Bad value: total_tax_amount when having tax with decimals for specific State

### ISSUE THREAD

**Issue by NiklasHogefjord** [krokedil/klarna-checkout-for-woocommerce#126]
Example:

Set tax for _Country_ US to be 8%.
Set tax for _Country_ US and _State_ OH to be 5,75%.

This ends up in an error message _Bad value: total_tax_amount_ when updating the order (after the shipping address update JS event has fired). 

The iframe does still display the information from the previous update. No errors displayed in front-end. This leads to an order in Woo with order status _Pending_ and no customer data (since the KCO order id is no longer a valid order).

**Comment by NiklasHogefjord** [comment]
This issue seem to be two folded. 

We introduced support for fetching Billing and Shipping State during shipping address change in version 1.7.0 (https://github.com/krokedil/klarna-checkout-for-woocommerce/releases/tag/1.7.0). However - we did this after running WC()->cart->calculate_totals(). This is now solved in this commit: 94b6ec43b828bf1826ef5728ef78bdf101db4736..

When it comes to calculation of shipping tax rate it looks like we in som e cases rounded tax rate to the nearest integer (5.75% turned into 6%). This should be solved in e67e1bb2eefee313f511a34fb79fe86d22abb474.

So to summarize this issue - @kristibarbosky perhaps you can test the current Develop branch and see if these two commits solved the problem?

**Comment by kristibarbosky** [comment]
@NiklasHogefjord - testing looks good! order_id=13745c15-6984-4e85-83ee-722812eee464 (US KCOv3 on a SEK Swedish store)

**Comment by NiklasHogefjord** [comment]
Sounds great @kristibarbosky. Will follow up on the other problems you reported on via Slack.

**Comment by NiklasHogefjord** [comment]
Closing this issue. It should be working as expected in v1.7.6 (https://github.com/krokedil/klarna-checkout-for-woocommerce/releases/tag/1.7.6).
